### DISCUSSION TITLE
categorise status code and use enum/classes

### DISCUSSION THREAD

**Discussion by sahasrara62** [encode/starlette#2685] [Ideas] [answered]
As of now status code are written as simple constant object like HTTP_100_CONTINUE = 100 https://github.com/encode/starlette/blob/8e1fc9b3d1578ae7d4b03c9cb2338019106f8544/starlette/status.py , and there are so many of them exists which are not easy to remember and one need to see them in codebase. why cannot we categories them like mentioned in https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml, so making enum of them according to level there are in , by i mean,for 1xx status code make a class called `InformationStatusCode` and  have all 1xx mentioned status code in it . so this way when who ever using starletter can simply load the status from 1xx, 2xx , 3xx as `from starlette.status import InformationStatusCode/SucessStatusCode/...` and calling each code using . 

so example one of change it will bring in fastapi code https://github.com/fastapi/fastapi/blob/b203d7a15fb5b49635bd81811e09ad94700f68a6/fastapi/security/http.py#L199 as 

```
from starlette.status import ClientErrorStatusCode

# line 199 https://github.com/fastapi/fastapi/blob/b203d7a15fb5b49635bd81811e09ad94700f68a6/fastapi/security/http.py#L199

status_code = ClientErrorStatusCode.HTTP_401_UNAUTHORIZED
```
this will more simply the loading status code and categorise them

**Comment by Kludex** [comment, marked as answer]
Doesn't sound a valid reason. The current way is fine.
